File: csharp/DVSClient/DVSClient/server/address/layout/RestApiAddressLayoutAppliesTo.cs
using DVSClient.Address.Layout;
using Newtonsoft.Json;

namespace DVSClient.Server.Address.Layout
{
    public class RestApiAddressLayoutAppliesTo
    {
        [JsonProperty("country_iso")]
        public string? CountryIso { get; set; }

        [JsonProperty("datasets")]
        public IEnumerable<string>? Datasets { get; set; }

        [JsonProperty("language")]
        public string? Language { get; set; }

        [JsonProperty("script")]
        public string? Script { get; set; }

        public RestApiAddressLayoutAppliesTo() { }

        public RestApiAddressLayoutAppliesTo(AppliesTo layoutAppliesTo)
        {
            CountryIso = layoutAppliesTo.GetCountry()?.Iso3Code;
            Datasets = layoutAppliesTo.Datasets?.ToList().ConvertAll(dataset => dataset.DatasetCode);
            Language = layoutAppliesTo.Language;
            Script = layoutAppliesTo.Script;
        }
    }
}

File: typescript/src/server/address/layout/restApiAddressLayoutAppliesTo.ts
import { AppliesTo } from "../../../address/layout/appliesTo";

export type RestApiAddressLayoutAppliesTo = {
    country_iso?: string;
    datasets?: string[];
    language?: string;
    script?: string;
};


export function getApiRequestFromLayoutAppliesTo(appliesTo: AppliesTo): RestApiAddressLayoutAppliesTo  {
    const result: RestApiAddressLayoutAppliesTo = {};

    if (appliesTo.datasets && appliesTo.datasets.length > 0) {
        result.country_iso =  appliesTo.datasets[0].country.iso3Code;
        result.datasets = appliesTo.datasets.map(dataset => dataset.datasetCode);
    }
    if (appliesTo.language) {
        result.language = appliesTo.language;
    }
    if (appliesTo.script) {
        result.script = appliesTo.script;
    }
    return result;

}